src/main/scala/fi/vm/sade/hakurekisteri/integration/koski/KoskiArvosanaTrigger.scala

```diff
@@ -456,7 +461,8 @@ object KoskiArvosanaTrigger {
           }
 
           val laajuus = suoritus.koulutusmoduuli.laajuus.getOrElse(KoskiValmaLaajuus(None, KoskiKoodi("","")))
-          if((!isPakollinen || a2b2Kielet.contains(tunniste.koodiarvo)) && laajuus.yksikkö.koodiarvo == "3" && laajuus.arvo.getOrElse(BigDecimal(0)) < 2) {
+          if( isAikuistenPerusopetus && !isPakollinen && laajuus.yksikkö.koodiarvo != "3" ||
+              (!isPakollinen || a2b2Kielet.contains(tunniste.koodiarvo)) && laajuus.yksikkö.koodiarvo == "3" && laajuus.arvo.getOrElse(BigDecimal(0)) < 2) {
```

Comment:
Niin ja tässä tietty kans. Jos näitä päättyelyitä voisi yhdistää johonkin nimettyyn funkkariin tai muuttujaan tai useampiin, niin voisi olla hyvä. Nyt on vähän vaikeeta seurata logiikkaa.